Source/KSCrash/Reporting/Tools/KSHTTPMultipartPostBody.m

```diff
@@ -161,28 +163,34 @@ - (NSData*) data
     }
 
     NSMutableData* data = [NSMutableData dataWithCapacity:baseSize];
+    BOOL firstFieldSent = NO;
     for(KSHTTPPostField* field in _fields)
     {
-        [data appendUTF8Format:@"--%@\r\n", g_boundary];
+        if (firstFieldSent) {
+            [data appendUTF8Format:@"\r\n"];
+        } else {
+            firstFieldSent = YES;
+        }
+        [data appendUTF8Format:@"--%@\r\n", _boundary];
         if(field.filename != nil)
         {
             [data appendUTF8Format:@"Content-Disposition: form-data; name=\"%@\"; filename=\"%@\"\r\n",
-             [field.name URLEncoded],
-             [field.filename URLEncoded]];
+             field.name,
+             field.filename];
```

Comment:
OK. Let's do this:

```objc
[self toStringWithQuotesEscaped:field.name],
[self toStringWithQuotesEscaped:field.filename]];
```
...
```objc
- (NSString*) toStringWithQuotesEscaped:(NSString*) value
{
    return [value stringByReplacingOccurrencesOfString:@"\"" withString:@"\\\""];
}
```
That way the caller doesn't need to be aware of the inner implementation or idiosyncrasies of the spec, and can just trust that the class will do the right thing with any string passed in.